forecasttools/idata_w_dates_to_df.py

```diff
@@ -0,0 +1,127 @@
+"""
+Contains functions for converting
+Arviz InferenceData objects to Polars
+dataframes with dates and draws and
+for working with conversions of
+Polars dataframes to Tidy dataframes.
+"""
+
+from datetime import datetime
+
+import arviz as az
+import numpy as np
+import polars as pl
+
+
+def add_dates_as_coords_to_idata(
+    idata_wo_dates: az.InferenceData,
+    group_dim_dict: dict[str, str],
+    start_date_iso: str,
+) -> az.InferenceData:
+    """
+    Modifies the provided idata object by assigning
+    date arrays to each group specified in group_dim_dict.
+
+    Parameters
+    ----------
+    idata
+        The InferenceData object that contains
+        multiple groups (e.g., observed_data,
+        posterior_predictive).
+    group_dim_dict
+        A dictionary that maps InferenceData group
+        names (e.g., "observed_data", "posterior_predictive"
+        ) to dimension names (e.g., "obs_dim_0").
+    start_date_iso
+        The start date in ISO format (YYYY-MM-DD) from
+        which to begin the date range for the dimension.
+
+    Returns
+    -------
+    idata
+        The modified InferenceData object with
+        date coordinates assigned to each group.
+    """
+
+    # NOTE: failure mode is if groups don't have the same
+    # start date
+
+    # convert received start date string to datetime object
+    start_date_as_dt = datetime.strptime(start_date_iso, "%Y-%m-%d")
+    # copy idata object to avoid modifying the original
+    idata_w_dates = idata_wo_dates.copy()
+    # modify indices of each selected group to dates
+    for (
+        group_name,
+        dim_name,
+    ) in group_dim_dict.items():
+        idata_group = getattr(idata_w_dates, group_name, None)
+        if idata_group is not None:
+            interval_size = idata_group.sizes[dim_name]
+            interval_dates = (
+                pl.DataFrame()
+                .select(
+                    pl.date_range(
+                        start=start_date_as_dt,
+                        end=start_date_as_dt
+                        + pl.duration(days=interval_size - 1),
+                        interval="1d",
```

Comment:
Shouldn't this be configurable?

Reply:
`interval_size` or `interval`? Assuming `interval` since `interval_size` is determined by `idata_group.sizes[dim_name]`. If `interval`, then I agree without a `str` parameter so default is not `1D`.

Reply:
This is more relevant to #34